examples/ERP/plot_classify_EEG_quantum.py

```diff
@@ -0,0 +1,131 @@
+ERP EEG decoding with Quantum Classifier.
+====================================================================
+
+Decoding applied to EEG data in sensor space decomposed using Xdawn.
+After spatial filtering, covariances matrices are estimated, then projected in
+the tangent space and classified with a quantum SVM classifier.
+It is compared to the classical SVM.
+
+"""
+# Author: Gregoire Cattan
+# Modified from plot_classify_EEG_tangentspace.py
+# License: BSD (3-clause)
+
+import numpy as np
+
+from pyriemann.estimation import XdawnCovariances
+from pyriemann.tangentspace import TangentSpace
+from pyriemann_qiskit.classification import QuanticSVM
+
+import mne
+from mne import io
+from mne.datasets import sample
+
+from sklearn.pipeline import make_pipeline
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay
+from sklearn.base import TransformerMixin
+from matplotlib import pyplot as plt
+
+# cvxpy is not correctly imported due to wheel not building
+# in the doc pipeline
+__cvxpy__ = True
+try:
+    import cvxpy
+    del cvxpy
+except Exception:
+    __cvxpy__ = False
+
+print(__doc__)
+
+data_path = sample.data_path()
+
+###############################################################################
+# Set parameters and read data
+raw_fname = data_path + "/MEG/sample/sample_audvis_filt-0-40_raw.fif"
+event_fname = data_path + "/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif"
+tmin, tmax = -0.0, 1
+event_id = dict(aud_l=1, aud_r=2, vis_l=3, vis_r=4)
+
+# Setup for reading the raw data
+raw = io.Raw(raw_fname, preload=True, verbose=False)
+raw.filter(2, None, method="iir")  # replace baselining with high-pass
+events = mne.read_events(event_fname)
+
+raw.info["bads"] = ["MEG 2443"]  # set bad channels
+picks = mne.pick_types(
+    raw.info, meg=False, eeg=True, stim=False, eog=False, exclude="bads"
+)
+
+# Read epochs
+epochs = mne.Epochs(
+    raw,
+    events,
+    event_id,
+    tmin,
+    tmax,
+    proj=False,
+    picks=picks,
+    baseline=None,
+    preload=True,
+    verbose=False,
+)
+
+X = epochs.get_data()
+y = epochs.events[:, -1]
+
+# Reduce the number of classes as QuanticBase supports only 2 classes
+y[y % 3 == 0] = 0
+y[y % 3 != 0] = 1
+
+# Reduce trial number to dimish testing time
+X = X[:100]
+y = y[:100]
+
+# ...skipping the KFold validation parts (for the purpose of the test only)
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+
+###############################################################################
+# Decoding in tangent space with a quantum classifier
+
+# Time complexity of quantum algorithm depends on the number of trials and
+# the number of elements inside the correlation matrices
+# Thus we reduce elements number by using restrictive spatial filtering
+sf = XdawnCovariances(nfilter=1)
+
+
+# ...and dividing the number of remaining elements by two
+class Downsampler(TransformerMixin):
```

Comment:
Yes, features downsampling generates is rough and generates a lost of classif power.
My advice is to suppress it, and to apply a supervised or unsupervised dimension reduction.

Reply:
`XdawnCovariances(nfilter=1)` is a weird choice, because after this step, your covariance matrices are reduced to scalars. You should use `XdawnCovariances(nfilter=2)` or `=3` for example.

Moreover, `make_pipeline(sf, cov, tg, Downsampler(), qsvm)` is not OK, because `XdawnCovariances` applies the same processing as `ERPCovariances`. It should be `make_pipeline(cov, tg, qsvm)` or `make_pipeline(sf, tg, qsvm)`.

Reply:
On the figure above, I used these pipeline: `sf -> tg -> Downsampler -> qsvm` and `sf -> tg -> PCA -> qsvm`

Although `nfilter=1`, Xdawn should output 2 channels. 
![image](https://user-images.githubusercontent.com/6229031/137458790-fec54c77-3bbc-4563-9f8f-32e538e3d90c.png)
Unfortunately, I cannot go higher in a quantum computer simulator, as `nfilter=2` will result in a tangent vector of size 36, which is not supported by the quantum simulator :
![image](https://user-images.githubusercontent.com/6229031/137459598-8f0881e0-1e77-4696-bbef-eeef198d47a7.png)

Below is a confusion matrix with the same data. Actually, I am surprised, because (on this particular example) it does not perform as well as with the naive dimension reduction:

![no_dim_red_2channels](https://user-images.githubusercontent.com/6229031/137459687-cd6af92b-1016-4087-9295-c752e05e8ba0.png)

I agree with you though that this Downsampling si really rough and at the ends depends on the indices wich are selected (here below an example of Downsampling but with impair values):

![without_pca_indent](https://user-images.githubusercontent.com/6229031/137461254-0729ce42-e135-4b2e-8b6a-f6210a1a3505.png)

Reply:
Ok, I updated the readme as suggested.
Also, I ran a complete [test ](https://github.com/pyRiemann/pyRiemann-qiskit/files/7358211/compare_dim_red.txt) on "vis left vs right" with a 5-fold cross-validation pipeline on all trials. The size of the feature was reduced from 10 to 5. Results are below (top line is SVC and bottom line is QSVM, score is balanced accuracy):
![dim_red](https://user-images.githubusercontent.com/6229031/137600886-58555b5c-0071-4397-8565-e193073ada05.png)